src/test/unit/test-utils.ts

```diff
@@ -28,4 +29,31 @@ export function createSampleViolation(location: CodeLocation, ruleName: string =
         fixes: fixes,
         suggestions: suggestions
     };
+}
+
+// To help test asyncronous code (when we purposely do not await a promise), we can use this function to help wait for
+// async operation to take place successfully. If it doesn't within the specified timeout, then an exception is thrown.
+export async function expectEventuallyIsTrue(conditionFn: () => boolean, timeout: number = 5000, interval: number = 50): Promise<void> {
```

Comment:
This seems like a useful utility, but is it redundant with `node:util`'s `promisify`?